### ISSUE TITLE
CollectionView scrolling properties

### ISSUE THREAD

**Issue by IeuanWalker** [dotnet/maui#25292]
Trying to create a scrolldown indicator similar to what I have done for the scrollview - 
https://github.com/user-attachments/assets/2e023a5a-56a3-45ff-b082-6156500304d0

For the scrollview it was pretty straightforward logic - 
```csharp
scrollView.Scrolled += async (_, e) =>
{
	double spaceAvailableForScrolling = scrollView.ContentSize.Height - scrollView.Height;

	if(spaceAvailableForScrolling < e.ScrollY + scrollDownIndicatorBuffer)
	{
		// Hide scroll
		await Task.WhenAll(
			control.Indicator.FadeTo(0, easing: Easing.CubicInOut),
			control.Indicator.TranslateTo(0, 50, easing: Easing.CubicInOut));

		control.Indicator.IsVisible = false;
	}
	else
	{
		// Show scroll
		control.Indicator.IsVisible = true;

		await Task.WhenAll(
			control.Indicator.FadeTo(1, easing: Easing.CubicIn),
			control.Indicator.TranslateTo(0, -10, easing: Easing.CubicInOut));
	}
};
```

But CollectionView doesn't seam to have the same properties, and the properties it does have doesn't seam to add up -
```csharp
collectionView.Scrolled += async (_, e) =>
{
	double totalHeight = collectionView.Measure(double.PositiveInfinity, double.PositiveInfinity).Request.Height;
	double visibleHeight = collectionView.Height;
	double maxScrollOffset = totalHeight - visibleHeight;
	double currentScrollPosition = e.VerticalOffset;

	// Debug output
	System.Diagnostics.Debug.WriteLine($"Total Height: {totalHeight}, Visible Height: {visibleHeight}, Max Scroll Offset: {maxScrollOffset}, Current Scroll Position: {currentScrollPosition}");
};
```

which outputs (after scrolling to the bottom of the CollectionView)
```
 Total Height: 997.3333333333334, Visible Height: 512.6666666666666, Max Scroll Offset: 484.66666666666674, Current Scroll Position: 70.33333333333333
```

`e.VerticalOffset` never equals the maximum offset. My only hung is either `e.VerticalOffset` isn't calculated correctly, or the way I'm calculating the totalHeight is incorrect

_____

Basically, all I'm trying to do is determine if there are items that are off the screen (below, to scroll down too), so if there is any way to determine that, any guidencance would be great
